src/block_utils.jl

```diff
@@ -0,0 +1,57 @@
+
+# Reduced QR factorization with Householder reflections:
+# Q, R = householder(A)
+#
+# Input :
+# A an n-by-k matrix, n ≥ k
+#
+# Output :
+# Q an n-by-k orthonormal matrix: QᴴQ = Iₖ
+# R an k-by-k upper triangular matrix: QR = A
+function householder(A::AbstractMatrix{FC}) where FC <: FloatOrComplex
+  n, k = size(A)
+  Q = copy(A)
+  τ = zeros(FC, k)
+  R = zeros(FC, k, k)
+  householder!(Q, R, τ)
+end
+
+function householder!(Q::AbstractMatrix{FC}, R::AbstractMatrix{FC}, τ::AbstractVector{FC}) where FC <: FloatOrComplex
+  n, k = size(Q)
+  R .= zero(FC)
+  @kgeqrf!(Q, τ)
```

Comment:
I guess that `@kgeqrf!` and `@korgrq!` are defined in Krylov.jl? Genuine question: what is the motive to use a macro there?

Reply:
I use the macro to have a common dispatch between CPU / GPU. I also use it to sometimes to provide a buffer.